pkg/workloads/console/integration/integration_test.go

```diff
@@ -66,37 +88,211 @@ var _ = Describe("Console", func() {
 
 	Describe("Creating resources", func() {
 		It("Sets console.spec.user from rbac", func() {
+			consoleTemplate := &workloadsv1alpha1.ConsoleTemplate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-first-console-template",
+					Namespace: namespace,
+				},
+				Spec: workloadsv1alpha1.ConsoleTemplateSpec{
+					AdditionalAttachSubjects: []rbacv1.Subject{},
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								corev1.Container{
+									Image:   "alpine:latest",
+									Name:    "console-container-0",
+									Command: []string{"sleep", "100"},
+								},
+							},
+							RestartPolicy: "Never",
+						},
+					},
+				},
+			}
+
+			By("Creating console template")
+			Expect(mgr.GetClient().Create(context.TODO(), consoleTemplate)).NotTo(
+				HaveOccurred(), "failed to create Console Template",
+			)
+
 			csl := &workloadsv1alpha1.Console{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      "foo",
+					Name:      "my-first-console",
 					Namespace: namespace,
 				},
 				Spec: workloadsv1alpha1.ConsoleSpec{
-					User: "", // deliberately not configured, this should be set by the webhook
+					ConsoleTemplateRef: corev1.LocalObjectReference{Name: "my-first-console-template"},
```

Comment:
Nice Kube tekkers. I didn't know this existed when I did the drb, but it totally should be this given it opening the possibility of using apimachinery helpers

👌